drivers/flash/flash_stm32_qspi.c

```diff
@@ -55,6 +55,9 @@ LOG_MODULE_REGISTER(flash_stm32_qspi, CONFIG_FLASH_LOG_LEVEL);
 
 #if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_qspi_nor)
 
+/* In dual-flash mode, total size is twice the size of one flash component */
+#define STM32_QSPI_DOUBLE_FLASH	DT_PROP(DT_NODELABEL(quadspi), dual_flash)
```

Comment:
Wouldn't this better be added to the dev->cfg to allow a combination of `dual-flash` and `non-dual-flash` on different qspi ports ?

Reply:
The driver is for one qspi instance and the HW is _exclusively_ dual-flash or not (no board has qspi-controller connected to dual-flash and another qspi with single NOR qspi-flash
with a flash_stm32_qspi_config.dual_flash = DT_PROP(DT_NODELABEL(quadspi), dual_flash)
you still have the flash_stm32_qspi_config.flash_size affected by the DT_PROP(DT_NODELABEL(quadspi), dual_flash)
(flash_stm32_qspi_config is a const)

Reply:
I agree that there is no such board at the moment, but couldn't a stm32h750xb custom board be created that does have both dual and non-dual flash at the same time.

Yes the config for _.flash_size would also need to use the DT_PROP(..., dual flash).